features/0067-didcomms-diddoc-conventions/README.md

```diff
@@ -0,0 +1,180 @@
+
+Of particular interested to this RFC is the definition of [service endpoints](https://w3c-ccg.github.io/did-spec/#service-endpoints). The primary object of this RFC is to document the Did communication service type and describe the associated conventions.
+
+### Service Conventions
+
+As referenced above within the [Did specification](https://w3c-ccg.github.io/did-spec/#did-documents) lies a section called [service endpoints](https://w3c-ccg.github.io/did-spec/#service-endpoints), this section of the Did document is reserved for `any type of service the entity wishes to advertise, including decentralized identity management services for further discovery, authentication, authorization, or interaction`.
+
+When a Did document wishes to express support for Did communications, the following service definition is defined.
+
+```json
+{
+  "service": [{
+    "id": "did:example:123456789abcdefghi#did-communication",
+    "type": "did-communication",
+    "priority" : 0,
+    "recipientKeys" : [ "did:example:123456789abcdefghi#1" ],
+    "routingKeys" : [ "did:example:123456789abcdefghi#1" ],
+    "serviceEndpoint": "https://agent.example.com/"
+  }]
+}
+```
+
+- id : Required by the [Service Endpoints Spec](https://w3c-ccg.github.io/did-spec/#service-endpoints).
+- type : Required by the [Service Endpoints Spec](https://w3c-ccg.github.io/did-spec/#service-endpoints). It is mandatory that this field is set to `did-communication`
+- priority : This represents the priority of the service endpoint, used for distinction when multiple `did-communication` service endpoints are present in a single Did document. It is mandatory that this field is set to an unsigned integer with the default value of `0`.
+- recipientKeys : This is an array of [did key references](https://w3c-ccg.github.io/did-spec/#public-keys) used to denote the default recipients of an endpoint. (*note-1*)
+- routingKeys: This is an array of [did key references](https://w3c-ccg.github.io/did-spec/#public-keys), ordered from most destward to most srcward, used to denote the individual routing hops in between the sender and recipients. See [TBC]() for more information on how routing is intended to operate.
+- serviceEndpoint : Required by the [Service Endpoints Spec](https://w3c-ccg.github.io/did-spec/#service-endpoints). This URL based endpoint is used to declare how the message should be sent. Did communication is transport agnostic, and therefore leverages existing application level transport protocols. However for each transport defined, which is identified by the URL scheme e.g `http`, a set of transport specific considerations are defined see [transports](../0025-didcomms-transports/README.md) for more details.
+
+>Notes
+>1. The keys featured in this array must resolve to keys of the same type, for example a mix `Ed25519VerificationKey2018` or `RsaVerificationKey2018` in the same array is invalid.
```

Comment:
@tplooker why must they be of the same type?

Reply:
@llorllale please have a look at https://github.com/hyperledger/aries-rfcs/tree/master/features/0019-encryption-envelope basically with our current encryption envelope when you encrypt to multiple recipients, the outputed JWE structure assumes the recipient keys are of all the same type. Supporting multiple recipients encrypted in a common JWE using different encryption primatives is out of scope for JWE's I believe too. @kdenhartog perhaps you can provide some additional commentary to this point?

Reply:
The point I am trying to make is that if you resolve a service decleration from a did doc and prepare a message for that service it should result in one encrypted envelope destined to be sent to one endpoint.